doc/source/schema/schema.md

```diff
@@ -36,33 +39,73 @@ The terminology used in this specification is based on [xarray's terminology](ht
+* Dimensions in InferenceData (including sample dimensions) should be identified by name only.
+* For groups like `observed_data` or `constant_data`, all sample dimensions can be
+  omitted. For groups like `prior`, `posterior` or `posterior_predictive` either `sample` has to be
+  present or both `chain` and `draw` dimensions need to be present. Any combinations that follow
+  this are valid.
 * Dimensions must be named and share name with a coordinate specifying the index values, called coordinate values.
 * Coordinate values can be repeated and should not necessarily be numerical values.
 * Variables must not share names with dimensions.
-* Moreover, each group contains the following attributes:
-  - `created_at`: the date of creation of the group.
-  - `inference_library`: the library used to run the inference.
-  - `inference_library_version`: version of the inference library used.
+* Groups, variables or the InferenceData itself can have arbitrary metadata stored.
 
-### Relations
-`InferenceData` data objects contain any combination of the groups described below. There are also some relations (detailed below) between the variables and dimensions of different groups. Hence, whenever related groups are present they should comply with these relations.
+### Metadata
+No metadata is _required_ to be present in order to be compliant with the InferenceData schema.
+However, it is recommended to store the following fields when relevant:
+* `name`: InferenceData objects represent multiple quantities related to Bayesian modelling,
+  but they are all tied to a single model. The model identifier can be added as metadata
+  to simplify the calls to model comparison functions.
+* `created_at`: the date of creation of the group.
+* `arviz_version`: the version of the ArviZ library that generated the InferenceData
+* `arviz_language`: the programming language from which ArviZ was used to create the InferenceData
+* `inference_library`: the library used to run the inference.
+* `inference_library_version`: version of the inference library used.
+
+
+### Relations between groups
+`InferenceData` data objects contain any combination of the groups described below. There are also some relations (detailed below) between the variables and dimensions of different groups. Hence, whenever related groups are present they should comply with these relations. Neither the presence of groups nor described below or the lack of some of the groups described below go against the schema.
 
 #### `posterior`
-Samples from the posterior distribution p(theta|y).
+Samples from the posterior distribution p(theta|y) in the parameter (also called constrained) space.
+
+(schema/unconstrained_posterior)=
+#### `unconstrained_posterior`
+Samples from the posterior distribution p(theta_transformed|y) in the unconstrained (also called transformed) space.
+
+Only variables that undergo a transformation for sampling should be present here.
```

Comment:
This might be a problematic for Stan where it is easy to calculate the transformation but hard to make sure that transformation is not unity (or is comparing values against each other valid way to do this?)

Reply:
This is only relative to the samples, so comparing values against each other is perfeclty fine. Even in edgecases like a transformation that is the unity in most of the space but not everywhere, if all the samples are the same then the transformation is not doing anything, we don't need to save the samples in both places; and if we generate new posterior samples, we also need to generate the associated unconstrained samples both of which are generated by the PPLs what will know if and which transformations to do.

Maybe we can write it in a bit more clear phrasing. Not the good style for the doc, but in a conversational style it could be something like: I have the posterior samples and I want to have the unconstrained samples. What are the variables whose
samples I need to replace with different ones? -> those variables and samples are the ones that go in this `unconstrained_sample` group.